test_fms/diag_manager/test_modern_diag.F90

```diff
@@ -108,20 +131,25 @@ program test_modern_diag
 id_ug = diag_axis_init("grid_index",  real(ug_dim_data), "none", "U", long_name="grid indices", &
                          set_name="land", DomainU=land_domain, aux="geolon_t geolat_t")
 
-id_z  = diag_axis_init('z',  z,  'point_Z', 'z', long_name='point_Z')
+id_z2 = diag_axis_init('z_edge',  z,  'point_Z', 'z', long_name='point_Z')
+id_z  = diag_axis_init('z',  z,  'point_Z', 'z', long_name='point_Z', edges = id_z2)
+
 call diag_axis_add_attribute (id_z, 'formula', 'p(n,k,j,i) = ap(k) + b(k)*ps(n,j,i)')
 call diag_axis_add_attribute (id_z, 'integer', 10)
 call diag_axis_add_attribute (id_z, '1d integer', (/10, 10/))
 call diag_axis_add_attribute (id_z, 'real', 10.)
 call diag_axis_add_attribute (id_x, '1d real', (/10./))
 call diag_axis_add_attribute (id_ug, 'compress', 'x y')
 
-if (id_x  .ne. 1) call mpp_error(FATAL, "The x axis does not have the expected id")
-if (id_y  .ne. 2) call mpp_error(FATAL, "The y axis does not have the expected id")
-if (id_x3 .ne. 3) call mpp_error(FATAL, "The x3 axis does not have the expected id")
-if (id_y3 .ne. 4) call mpp_error(FATAL, "The y3 axis does not have the expected id")
-if (id_ug .ne. 5) call mpp_error(FATAL, "The ug axis does not have the expected id")
-if (id_z  .ne. 6) call mpp_error(FATAL, "The z axis does not have the expected id")
+if (.not. debug) then
```

Comment:
What is `debug` supposed to do?  Why are these checks in here if they are just being skipped?  Are the test specific?

Reply:
`debug` is a namelist flag. when `.true.` it ignored the ids checks. 
https://github.com/NOAA-GFDL/FMS/blob/488dfa195f0f9abb2fb4c9e74d63f14b00b92be0/test_fms/diag_manager/test_modern_diag.F90#L81-L84

This is going to be useful when debugging because I will be able to modify the diag_table.yaml to just include whatever feature I am debugging (like static fields, or diurnal fields, etc)